packages/scope-manager/tests/fixtures/type-declaration/mapped-named.ts.shot

```diff
@@ -13,7 +13,7 @@ ScopeManager {
       ],
```

Comment:
i realise now that our tests for named mapped types is lacking.
Could you please add the following two cases:

```ts
type T = 1; // should have 0 references
type M = { [T in string as T]: 1 };
```

```ts
type T = 1; // should have 1 reference
type M = { [K in string as T]: 1 };
```